pkg/proc/stack.go

```diff
@@ -455,12 +475,15 @@ func (it *stackIterator) advanceRegs() (callFrameRegs op.DwarfRegisters, ret uin
 				it.err = err
 			} else {
 				ret = reg.Uint64Val
+				if isARM64 && it.bi.GOOS == "linux" {
```

Comment:
This shouldn't be done here but in (*ARM64).SwitchStack (ret is saved in it.frame.Ret).

Reply:
If the code is put into (*ARM64).SwitchStack, the backtrace no longer works.  In particular, the "retaddr = uint64(it.regs.CFA + regRule.Offset)" assignment in "advanceRegs" computes the wrong value, and it seems tricky to correct this value later.

Reply:
If I'm not mistaken, retaddr gets stored in addrret and only used to do a check related to stack barriers, a feature that isn't present in recent versions of Go and wouldn't matter in the case of exception frames anyway.

Reply:
I ran a few more checks, and still can't get the code to work when it is placed in SwitchStack.  I've got some other work to do for the next few days, but should be able to get back to this early next week.  I'll keep you posted.